internal/services/gateway/action/runwebhookdelivery.go

```diff
@@ -82,3 +82,29 @@ func (h *ActionHandler) GetProjectRunWebhookDeliveries(ctx context.Context, req
 
 	return res, nil
 }
+
+type ProjectRunWebhookRedeliveryRequest struct {
+	ProjectRef           string
+	RunWebhookDeliveryID string
+}
+
+func (h *ActionHandler) ProjectRunWebhookRedelivery(ctx context.Context, req *ProjectRunWebhookRedeliveryRequest) error {
+	project, _, err := h.configstoreClient.GetProject(ctx, req.ProjectRef)
+	if err != nil {
+		return util.NewAPIError(util.KindFromRemoteError(err), err)
+	}
+	isUserOwner, err := h.IsAuthUserProjectOwner(ctx, project.OwnerType, project.OwnerID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to determine permissions")
+	}
+	if !isUserOwner {
+		return util.NewAPIError(util.ErrForbidden, errors.Errorf("user not authorized"))
+	}
+
+	_, err = h.notificationClient.RunWebhookRedelivery(ctx, req.RunWebhookDeliveryID)
```

Comment:
the gateway api requires the project id but the notification api doesn't require it. You are checking that the user can do the action on the provided project id but you aren't checking that the delivery belongs to that project causing a security issue.

The tests should also cover such case to avoid regressions.

Reply:
I can change ta actual redelivery api(notificaiton) to `projects/{projectid}/runwebhookdeliveries/{runwebhookdeliveryid}/redelivery` the notification will return a notexists error if the delivery not belongs to the project.

Or I can add in notification(in a new PR) an api to get the runwebhookdelivery by id, the gateway check that the delivery belongs to the project.

What do you think is the better way?

Reply:
> I can change ta actual redelivery api(notificaiton) to `projects/{projectid}/runwebhookdeliveries/{runwebhookdeliveryid}/redelivery` the notification will return a notexists error if the delivery not belongs to the project.

Notification currently has a single api to retrieve the deliveries by project id. So it makes sense to also do the redelivery by project id.

The tests for both notification and gateway should cover the case of a redelivery id not belonging to the project.